backend/src/main/java/hu/flowacademy/worksheet/service/KeycloakClientService.java

```diff
@@ -1,61 +1,51 @@
 import org.keycloak.admin.client.resource.RolesResource;
 import org.keycloak.admin.client.resource.UserResource;
 import org.keycloak.admin.client.resource.UsersResource;
 import org.keycloak.representations.idm.CredentialRepresentation;
 import org.keycloak.representations.idm.RoleRepresentation;
 import org.keycloak.representations.idm.UserRepresentation;
-import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
 import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
 
 import javax.ws.rs.core.Response;
+import java.net.URI;
 import java.util.Arrays;
 import java.util.Locale;
 
+@RequiredArgsConstructor
 @Service
 public class KeycloakClientService {
 
-    @Value("${keycloakBackendClient.server-url}")
-    private String SERVER_URL;
-    @Value("${keycloakBackendClient.realm}")
-    private String REALM;
-    @Value("${keycloakBackendClient.realm2}")
-    private String REALM2;
-    @Value("${keycloakBackendClient.adminusername}")
-    private String ADMIN_USERNAME;
-    @Value("${keycloakBackendClient.adminpassword}")
-    private String ADMIN_PASSWORD;
-    @Value("${keycloakBackendClient.client-id}")
-    private String CLIENT_ID;
-    private String USER_ROLE = "worksheetuser";
+    private RestTemplate restTemplate = new RestTemplate();
+
+    private final KeycloakClientConfiguration keycloakClientConfiguration;
+    private final Keycloak keycloak;
 
-    private Keycloak getInstance() {
-        return KeycloakBuilder
-                .builder()
-                .serverUrl(SERVER_URL)
-                .realm(REALM)
-                .username(ADMIN_USERNAME)
-                .password(ADMIN_PASSWORD)
-                .clientId(CLIENT_ID)
-                .build();
-    }
 
     public int createAccount(String name, String email) throws WorksheetUserException {
         CredentialRepresentation credential = createCredentials();
         UserRepresentation user = createUserRepresentation(name, email, credential);
-        RealmResource ourRealm = getInstance().realm("worksheet");
+        RealmResource ourRealm = keycloak.realm("worksheet");
         RolesResource roleList = ourRealm.roles();
         UsersResource everyOne = ourRealm.users();
         RoleRepresentation roleToUse = roleList.get("worksheetuser").toRepresentation();
-        javax.ws.rs.core.Response response = getInstance().realm("worksheet").users().create(user);
+        javax.ws.rs.core.Response response = keycloak.realm("worksheet").users().create(user);
         String userId = CreatedResponseUtil.getCreatedId(response);
         UserResource oneUser = everyOne.get(userId);
         oneUser.roles().realmLevel().add(Arrays.asList(roleToUse));
```

Comment:
Amennyiben List-et vár az add paraméterként, használj inkább List.of(...)-ot Arrays.asList helyett.